docs/DeveloperGuide.md

```diff
@@ -1,33 +1,230 @@
+within the ActiveEdge application, promoting modularity and separation of concerns.
+
+![Command](../images/Command-Class.png)
+
+
+
+
+### Storage
+Storage is the main class responsible for file operations and data management.
+UserDetailsList, LogHeight, and LogWeight handle user details such as height and weight logs.
+TaskList manages various types of tasks including goals, meals, water intake, and exercises.
+GoalTask, MealTask, WaterTask, and ExerciseTask are specific task types with their attributes.
+AddBMICommand, AddHeightCommand, and AddWeightCommand are commands for adding BMI, height, and weight respectively.
+The relationships depicted in the diagram are primarily composition and inheritance:
+
+
+![Main](../images/Storage.png)
+
+Storage has composition relationships with UserDetailsList and TaskList as it manages instances of these classes.
+UserDetailsList and TaskList have a composition relationship with their respective contained classes (LogHeight, LogWeight, GoalTask, MealTask, WaterTask, ExerciseTask) 
+as they hold lists of instances of these classes. AddBMICommand, AddHeightCommand, and AddWeightCommand are standalone commands 
+used within the Storage class but don't directly interact with the other classes in the diagram.
+
+### Main Component
+The ActiveEdge class serves as the main entry point for the ActiveEdge application, handling user input parsing and interaction through a command-line interface.
+![Main Component](../images/Main Component.png)
 
-## Design & implementation
+### UI Package
+The activeedge.ui package encapsulates the user interface components of the ActiveEdge application, facilitating interactions such as printing welcome messages, 
+logging meal and exercise data, displaying goal information, providing help instructions, and managing user queries and commands for health tracking and goal setting.
+![Ui](../images/Ui Package.png)
 
+Under commands, there are 17 sub-components:
+```AddBMICommand```: A command responsible for calculating and adding Body Mass Index (BMI) information.
+```AddGoalsCommand```: A command responsible for users to set and track their fitness goals.
+```AddHeightCommand```: A command responsible for users to input and track their height information for comprehensive health tracking.
+```AddWeightCommand```: A command responsible for users to input and track their weight measurements to monitor their progress and achievements.
+```ClearCommand```: A command responsible for users to clear or delete entries, such as meals, from their tracking history.
+```DeleteTaskCommand```: A command responsible for users to delete or remove exercise tasks.
+```FindCommand```: A command responsible for users to search specific keywords.
+```ShowSummaryCommand```: A command responsible for users to view a summary or overview of their calorie intake,calorie burnt during exercise .
+```ViewWaterIntakeCommand```: A command responsible for users to view and track their water intake levels.
+
```

Comment:
While it is good to list out the components here, it is rather messy and confusing. Can consider listing out using bullet points!
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/64789669/bacf1090-6a20-4026-be0b-40f771f57c86)